crates/bevy_animation/src/lib.rs

```diff
@@ -124,10 +126,59 @@ impl Default for PlayingAnimation {
             elapsed: 0.0,
             animation_clip: Default::default(),
             path_cache: Vec::new(),
+            root_motion: None,
         }
     }
 }
 
+#[derive(Reflect, FromReflect, Default, Clone)]
+enum RootMotionMode {
+    #[default]
+    Enabled,
+    Locked,
+}
+
+/// Controls how root motion is handled during an animation
+#[derive(Reflect, Default, FromReflect, Clone)]
+pub struct RootMotion {
+    mode: RootMotionMode,
```

Comment:
Couldn't this be a bool called locked? Or is it expected that users will match on the mode?